docs/adr/0008-wrapper-code.md

```diff
@@ -0,0 +1,58 @@
+# Wrapper code
+
+* Status: draft
+* Deciders: 
+* Date: ???
+
+## Context and Problem Statement
+
+Application-services components currently consist of a Rust core that gets wrapped by Swift and Kotlin wrappers.
+In the past, these wrappers were strictly necessary because of deficiencies in our FFI strategy.
+However, UniFFI is reaching the point where it can support all our requirements and it's possible to remove the wrapper code.
+
+So... should we?
+
+### Scope
+
+This ADR discusses what our general policy on wrapper code should be.
+It does not cover how we should plan our work.
+If we decide to reject wrapper code, we do not need to commit to any particular timeline for actually removing it.
+
+## Decision Drivers
+
+## Considered Options
+
+* **(A) Embrace wrapper code and continue to use it**
+* **(B) Reject wrapper code and remove it**
+* **(C) Use wrapper code for async wrapping only**
+
+## Decision Outcome
+
+## Pros and Cons of the Options
+
+### (A) Embrace wrapper code and continue to use it
+
+* Good, because wrapper code allows us to wrap sync Rust code in async wrapper functions and this is a low-risk approach to async.
+* Good, because wrapper code allows us to mitigate breaking changes in the Rust code.
+* Bad, because there are better ways to handle breaking changes.
+  We could introduce breaking changes, and large changes in general, behind a Rust feature flag.
+  We would wait to enable the feature flag on the megazord until consumer application was ready.
```

Comment:
This is my current feeling, but I'm not that confident in it.  What do others think?

Reply:
What would this feature flag infrastructure look like? One the one hand I could see it being hugely beneficial for a lot of scenarios including releases backend logic for consumer features when the frontend isn't in place. On the other hand, this could get onerous to maintain with different consumers depending on the situation.